h2/src/main/org/h2/value/Transfer.java

```diff
@@ -530,6 +530,10 @@ public void writeValue(Value v) throws IOException {
             }
             break;
         default:
+            if (JdbcUtils.customDataTypesHandler != null) {
+                writeBytes(JdbcUtils.customDataTypesHandler.convert(v, Value.BYTES).getBytesNoCopy());
```

Comment:
Here we have to just do `writeBytes(v.getBytesNoCopy())`. It must do the same trick by default, but can have a more effective implementation inside.